### PR TITLE
Hong

### PR DESCRIPTION
**Pull request by chloe64**
add get_recipe and log_out

### CODE REVIEW

File: src/use_case/get_recipe/GetRecipeInteractor.java [hunk 1]
```diff
@@ -12,16 +14,17 @@ public GetRecipeInteractor(GetRecipeDataAccessInterface recipeDataAccessObject,
...
-//            recipePresenter.prepareFailView("Recipe not found.");
-//        } else {
-//            Folder folder = folderFactory.create(createFolderInputData.getFoldername());
-//            folderDataAccessObject.save(folder);
-//
-//            CreateFolderOutputData createFolderOutputData = new CreateFolderOutputData(folder.getName(), false);
-//            folderPresenter.prepareSuccessView(createFolderOutputData);
-//        }
+    public void execute(GetRecipeInputData getRecipeInputData) throws IOException {
+        if (false) {
```
Comment (yuxizzz) [inline src/use_case/get_recipe/GetRecipeInteractor.java]: why is there a false statement here?

File: src/view/UserRecipeView.java [hunk 2]
```diff
@@ -1,4 +1,24 @@
...
+import javax.swing.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+
+//TODO: nothing here
+
+public class UserRecipeView extends JPanel implements ActionListener, PropertyChangeListener {
+    public final String viewName = "recipe";
```
Comment (yuxizzz) [inline src/view/UserRecipeView.java:12]: why is there no information?

File: src/view/RecipeView.java [hunk 3]
```diff
@@ -1,4 +1,49 @@
...
+        this.add(buttons);
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+
+    }
+
+    @Override
+    public void propertyChange(PropertyChangeEvent evt) {
```
Comment (yuxizzz) [inline src/view/RecipeView.java:46]: why is no implementation?

**Comment by chloe64** [comment]
use_case and interface_adapter of get recipe; interface_adapter (except controller) of log out

File: src/interface_adapter/logout/LogoutState.java [hunk 4]
```diff
@@ -1,16 +1,20 @@
...
     public LogoutState() {
     }
-    public Integer getUserID() {
-        return userID;
+    public String getUserID() {
+        return username;
     }
-    public void setUserID(Integer userID) {
-        this.userID = userID;
+    public void setUserID(String username) {
```
Comment (yuxizzz) [inline src/interface_adapter/logout/LogoutState.java:14]: why is it setUserID?

Comment (chloe64) [reply src/interface_adapter/logout/LogoutState.java]: fixed

File: src/use_case/get_recipe/GetRecipeInteractor.java (hunk 1 above)
Comment (chloe64) [reply src/use_case/get_recipe/GetRecipeInteractor.java]: updated

**Pull request merged**